utils/src/main/scala/com/salesforce/op/utils/text/TextUtils.scala

```diff
@@ -30,23 +30,47 @@
 
 package com.salesforce.op.utils.text
 
+import org.json4s.DefaultFormats
+import org.json4s.jackson.Serialization
+
 
 case object TextUtils {
+  val defaultCleanParams = CleanTextParams(ignoreCase = true, cleanPunctuations = true)
 
-  def cleanOptString(raw: Option[String], splitOn: String = " "): Option[String] =
-    raw.map(t => cleanString(t, splitOn))
+  def cleanOptString(raw: Option[String], splitOn: String = " ",
+    cleanTextParams: CleanTextParams = defaultCleanParams): Option[String] =
+    raw.map(t => cleanString(t, splitOn, cleanTextParams))
 
-  def cleanString(raw: String, splitOn: String = " "): String = {
-    raw
-      .toLowerCase
+  def cleanString(raw: String, splitOn: String = " ", cleanTextParams: CleanTextParams = defaultCleanParams): String = {
+    val l = if (cleanTextParams.ignoreCase) raw.toLowerCase else raw
```

Comment:
looks like JDK has an interesting twist on ignoreCase and actually prefers to normalize to the upper case unless for the Georgian alphabet http://hg.openjdk.java.net/jdk7u/jdk7u6/jdk/file/8c2c5d63a17e/src/share/classes/java/lang/String.java#l1356
Consider more explicit flags: lowerCase, upperCase instead of more general ignoreCase